packages/cloud-cognitive/src/components/Datagrid/Datagrid/addons/Filtering/FilterFlyout.js

```diff
@@ -0,0 +1,137 @@
+// @flow
+/*
+ * Licensed Materials - Property of IBM
+ * 5724-Q36
+ * (c) Copyright IBM Corp. 2022
+ * US Government Users Restricted Rights - Use, duplication or disclosure
+ * restricted by GSA ADP Schedule Contract with IBM Corp.
+ */
+
+import React, {
+  useState,
+  useCallback,
+  useRef,
+  useMemo,
+  useContext,
+} from 'react';
+import PropTypes from 'prop-types';
+import { Button } from 'carbon-components-react';
+import { Filter16 } from '@carbon/icons-react';
+import cx from 'classnames';
+import { ActionSet } from '../../../../ActionSet';
+import { pkg } from '../../../../../settings';
+import { BATCH, INSTANT } from './constants';
+import { useClickOutside } from '../../../../../global/js/hooks';
+import { FilterContext } from './FilterProvider';
+
+const blockClass = `${pkg.prefix}--datagrid`;
+const componentClass = `${blockClass}-filter-flyout`;
+
+const FilterFlyout = ({
+  children,
+  updateMethod = BATCH,
+  title = 'Filter',
+  shouldClickOutsideToClose = false,
+  setAllFilters,
+}) => {
+  /** Refs */
+  const filterFlyoutRef = useRef(null);
+  const { filtersObjectArray } = useContext(FilterContext);
+
+  /** State */
+  const [open, setOpen] = useState(false);
+
+  /** Memos */
+  const showActionSet = useMemo(() => updateMethod === BATCH, [updateMethod]);
+
+  /** Methods */
+  const openFlyout = () => setOpen(true);
+  const closeFlyout = () => setOpen(false);
+
+  const onApply = useCallback(() => {
+    setAllFilters(filtersObjectArray.current);
+    closeFlyout();
+  }, [setAllFilters, filtersObjectArray]);
+
+  /** Effects */
+  useClickOutside(filterFlyoutRef, (target) => {
+    if (shouldClickOutsideToClose === false) {
+      return;
+    }
+
+    const hasClickedOnDatePicker = target.closest('.flatpickr-calendar');
+
+    if (!open || hasClickedOnDatePicker) {
+      return;
+    }
+
+    setOpen(false);
+  });
+
+  const renderActionSet = useCallback(() => {
+    return (
+      showActionSet && (
+        <ActionSet
+          actions={[
+            {
+              key: 1,
+              kind: 'primary',
+              label: 'Apply',
```

Comment:
Can we create a property that can be passed in for this in order to support translation?